Source/Csla.Shared/Reflection/ServiceProviderMethodCaller.cs

```diff
@@ -61,16 +64,16 @@ public static System.Reflection.MethodInfo FindDataPortalMethod(Type targetType,
         var factoryType = Csla.Server.FactoryDataPortal.FactoryLoader.GetFactoryType(factoryInfo.FactoryTypeName);
         if (factoryType != null)
         {
-          if (attributeType == typeof(CreateAttribute))
-            candidates = factoryType.GetMethods(_bindingAttr).Where(m => m.Name == factoryInfo.CreateMethodName).ToList();
-          else if (attributeType == typeof(FetchAttribute))
-            candidates = factoryType.GetMethods(_bindingAttr).Where(m => m.Name == factoryInfo.FetchMethodName).ToList();
-          else if (attributeType == typeof(DeleteAttribute))
-            candidates = factoryType.GetMethods(_bindingAttr).Where(m => m.Name == factoryInfo.DeleteMethodName).ToList();
-          else if (attributeType == typeof(ExecuteAttribute))
-            candidates = factoryType.GetMethods(_bindingAttr).Where(m => m.Name == factoryInfo.ExecuteMethodName).ToList();
+          if (typeof(T) == typeof(CreateAttribute))
```

Comment:
Grab the value of `typeof(T)` and use that in all the conditional checks.

Reply:
Will do